src/main/java/org/zalando/nakadi/service/timeline/TimelineSync.java

```diff
@@ -36,14 +36,15 @@
      * @param timeoutMs Timeout for sync operation.
      * @throws IllegalStateException In case when timeline update already started for specified event type
      */
-    void startTimelineUpdate(String eventType, long timeoutMs) throws InterruptedException, IllegalStateException;
+    void startTimelineUpdate(String eventType, long timeoutMs)
+            throws InterruptedException, IllegalStateException, RuntimeException;
 
     /**
      * Release publishing lock to event type
      *
      * @param eventType Event type to unlock publishing to.
      */
-    void finishTimelineUpdate(String eventType) throws InterruptedException;
+    void finishTimelineUpdate(String eventType) throws InterruptedException, RuntimeException;
```

Comment:
I don't think that it makes much sense to declare throwing of general RuntimeException which actually can be thrown by any method.